tests/nested/manual/remodel-simple/task.yaml

```diff
@@ -0,0 +1,106 @@
+  NESTED_CUSTOM_AUTO_IMPORT_ASSERTION: $TESTSLIB/assertions/test-snapd-remodel-auto-import.assert
+  NESTED_IMAGE_ID: uc20-remodel-testing
+  NESTED_ENABLE_TPM: false
+  NESTED_ENABLE_SECURE_BOOT: false
+  NESTED_USE_CLOUD_INIT: false
+  SNAPD_TEST_STORE: test-snapd
+
+prepare: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB"/nested.sh
+    UBUNTU_IMAGE="$GOHOME"/bin/ubuntu-image
+    variant=18
+    if os.query is-focal; then
+        variant=20
+    fi
+    IMAGE_NAME="$(nested_get_image_name core)"
+    # fetch the snapd snap from edge and repack the deb into the snap
+    snap download --channel="latest/edge" snapd
+    "$TESTSTOOLS"/snaps-state repack_snapd_deb_into_snap snapd
+
+    snap pack "$TESTSLIB/snaps/test-snapd-remodel-pc-$variant"
+    UBUNTU_STORE_AUTH_DATA_FILENAME="$TESTSLIB/remodel-store-viewer.auth" \
+        "$UBUNTU_IMAGE" snap "$TESTSLIB/assertions/test-snapd-remodel-pc-$variant.model" \
+            --snap test-snapd-remodel-pc_*.snap \
+            --snap "$PWD/snapd-from-deb.snap" \
+            --output-dir "$NESTED_IMAGES_DIR"
+    # ubuntu-image dropped the --output parameter, so we have to rename
+    # the image ourselves, the images are named after volumes listed in
+    # gadget.yaml
+    find "$NESTED_IMAGES_DIR/" -maxdepth 1 -name '*.img' | while read -r imgname; do
+        if [ -e "$NESTED_IMAGES_DIR/$IMAGE_NAME" ]; then
+            echo "Image $IMAGE_NAME file already present"
+            exit 1
+        fi
+        mv "$imgname" "$NESTED_IMAGES_DIR/$IMAGE_NAME"
+    done
+
+    nested_create_assertions_disk
+    nested_start_core_vm
+
+restore: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB"/nested.sh
+    nested_shutdown
+
+execute: |
+    # shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB"/nested.sh
+
+    variant=18
+    if os.query is-focal; then
+        variant=20
+    fi
+
+    boot_id="$(tests.nested boot-id)"
+
+    tests.nested exec "snap model --assertion" | MATCH "brand-id: $SNAPD_TEST_STORE\$"
+    tests.nested exec "snap model --assertion" | MATCH '^model: test-snapd-remodel-pc$'
+    tests.nested exec "snap model --assertion --serial" | MATCH 'serial: 1234-snapd-remodel-testing$'
+
+    # none of the new snaps are installed yet
+    not tests.nested exec "snap list hello-world"
+    not tests.nested exec "snap list core"
+
+    # snapd gets stuck ensuring prerequisites of hello-world, which is the
+    # installation of core, workaround this problem by installing core
+    # explicitly
+    # TODO: drop this once prereq is fixed
+    tests.nested exec "snap install core"
+
+    tests.nested copy "$TESTSLIB/assertions/test-snapd-remodel-pc-just-model-$variant.model"
+    CHANGE_ID="$(tests.nested exec "sudo snap remodel --no-wait test-snapd-remodel-pc-just-model-$variant.model")"
+    test -n "$CHANGE_ID"
+    # very long retry wait for the change to be in stable state, once it's
+    # stable it does not mean that the change was successful yet
```

Comment:
Nitpick: You mean it may fail, but we match `Done|Undone|Error` not to delay the test if something fails (and we just fail ASAP by matching "Done" below? I'd consider dropping "yet" from the comment.